File: java/micronauts/src/main/java/com/benchmark/domain/BenchmarkController.java
package com.benchmark.domain;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.QueryValue;
import io.micronaut.http.HttpResponse;
@Controller("/benchmark")
public class BenchmarkController {

    private double compute(int n) {
        double result = 0.0;
        double[] temp = new double[n];

        for (int i = 0; i < n; i++) {
            temp[i] = Math.sqrt(i * i + i);
            result += temp[i];
        }

        return result;
    }

    @Get
    public HttpResponse<String> get(@QueryValue Integer n) throws Exception {
        double result = compute(n);

        return HttpResponse.ok("OK")
                .header("X-Benchmark-Result", Double.toString(result));
    }

}

File: nodejs/express/src/domain/benchmark.controller.ts
import { Controller, Get, Query, Res } from '@nestjs/common';
import { Response } from 'express'
@Controller('benchmark')
export class BenchmarkController {
    compute(n: number): number {
        let result = 0;
        const temp = new Array(n);

        for (let i = 0; i < n; i++) {
          temp[i] = Math.sqrt(i * i + i);
          result += temp[i];
        }

        return result;
    }

    @Get()
    async createUser(@Query('n') n: number, @Res({ passthrough: true }) res: Response) : Promise<string> {
        const result = this.compute(n);

        res.header('X-Benchmark-Result', result.toString());
    
        return "OK";
    }
}
